RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronCoreProducer.cc

```diff
@@ -0,0 +1,64 @@
+#include "RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronCoreProducer.h"
+#include "DataFormats/EgammaCandidates/interface/GsfElectronCore.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+#include "DataFormats/ParticleFlowReco/interface/GsfPFRecTrack.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include <map>
+
+using namespace reco ;
+
+LowPtGsfElectronCoreProducer::LowPtGsfElectronCoreProducer( const edm::ParameterSet& config )
+  : GsfElectronCoreBaseProducer(config)
+{
+  superClusters_ = consumes<reco::SuperClusterCollection>(config.getParameter<edm::InputTag>("superClusters"));
+  superClusterRefs_ = consumes< edm::ValueMap<reco::SuperClusterRef> >(config.getParameter<edm::InputTag>("superClusters"));
+}
+
+LowPtGsfElectronCoreProducer::~LowPtGsfElectronCoreProducer() {}
+
+void LowPtGsfElectronCoreProducer::produce( edm::Event& event, const edm::EventSetup& setup ) {
+
+  // Output collection
+  auto electrons = std::make_unique<GsfElectronCoreCollection>();
+
+  // Init
+  GsfElectronCoreBaseProducer::initEvent(event,setup) ;
+  if ( !useGsfPfRecTracks_ ) { edm::LogError("useGsfPfRecTracks_ is (redundantly) set to False!"); }
+  if ( !gsfPfRecTracksH_.isValid() ) { edm::LogError("gsfPfRecTracks handle is invalid!"); }
+  if ( !gsfTracksH_.isValid() ) { edm::LogError("gsfTracks handle is invalid!"); }
+
+  edm::Handle<reco::SuperClusterCollection> superClusters;
+  event.getByToken(superClusters_,superClusters);
+  if ( !superClusters.isValid() ) { edm::LogError("Problem with superClusters handle"); }
+
+  edm::Handle< edm::ValueMap<reco::SuperClusterRef> > superClusterRefs;
+  event.getByToken(superClusterRefs_,superClusterRefs);
+  if ( !superClusterRefs.isValid() ) { edm::LogError("Problem with superClusterRefs handle"); }
+
+  // Create ElectronCore objects
+  for ( size_t ipfgsf = 0; ipfgsf < gsfPfRecTracksH_->size(); ++ipfgsf ) {
+
+    // Refs to GSF(PF) objects and SC
+    reco::GsfPFRecTrackRef pfgsf(gsfPfRecTracksH_, ipfgsf);
+    reco::GsfTrackRef gsf = pfgsf->gsfTrackRef();
+    const reco::SuperClusterRef sc = (*superClusterRefs)[pfgsf];
+
+    // Construct and keep ElectronCore if GSF(PF) track and SC are present
+    GsfElectronCore* core = new GsfElectronCore(gsf);
```

Comment:
this gets leaked if its tracker driven seed. No need to be a pointer. Just make it regular object and pass its address to GsfElectronCoreBaseProducer::fillElectronCore() (which should be taking a reference but not a pointer, sigh). The "electrons->push_back(*core)" makes a copy and does not transfer ownership (its a std::vector<reco::GsfElectronCore>).

Reply:
Keep as a pointer, but delete at the end of the method, [here](https://github.com/CMSBParking/cmssw/pull/3/commits/4a918b28a19d2135595a29df07fb757cf21346e1#diff-75b29902f47a158fbc16fea507ba10cfR60). The same approach (using a pointer) is taken in (GED)GsfElectron(Core)Producer, although the GsfElectronCoreProducer does not delete the pointer ... Ok?